shared/src/test/scala/fr/hmil/roshttp/HttpRequestSpec.scala

```diff
@@ -654,5 +654,25 @@ object HttpRequestSpec extends TestSuite {
         }
       }
     }
+
+    "Cookies" - {
+      "created on and returned from server" - {
+        HttpRequest(s"$SERVER_URL/send_cookie")
+          .withCredentials(true)
+          .send()
+          .map(res => {
+            assert(res.statusCode == 200)
+          })
+      }
+      "resent to server and processed accordingly" - {
+        HttpRequest(s"$SERVER_URL/receive_cookie")
+          .withCredentials(true)
+          .send()
+          .map(res => {
+            assert(res.statusCode == 200)
+          })
+      }
+    }
```

Comment:
These tests aren't isolated, and that will cause flakiness.

Instead of two, you should create a single test case which:
1. makes a first request to set a cookie
2. makes a second request to verify the cookie.

For added security, you should generate a unique ID in the test case, and use this ID in the cookie name so that you isolate the test from potential side effects of cookies surviving multiple test runs.

Reply:
DONE.